src/Components/Authentication/helpers.ts

```diff
@@ -34,3 +37,37 @@ export async function sendAuthData(
   const data = await res.json()
   return data
 }
+
+export const checkEmail = ({ values, actions, shouldExist }) => {
+  const query = email => `
+    query {
+      user(email: "${email}") {
+        userAlreadyExists
+      }
+    }
+  `
+  return metaphysics(
+    { query: query(values.email) },
+    {
+      appToken: sharify.data.XAPP_TOKEN,
+    }
+  ).then(({ data }: any) => {
+    if (data.user.userAlreadyExists) {
+      if (shouldExist) {
+        return true
+      } else {
+        actions.setFieldError("email", "Email already exists.")
+        actions.setSubmitting(false)
```

Comment:
Curious to know where these actions come from -- are they stored in context?

Reply:
These actions are passed down as args from Formik's `onSubmit` here: https://github.com/artsy/reaction/pull/750/files#diff-789fe1e619c39fe8bf9ce02131a25dddR26